### PR TITLE
Added username filtering to catalog list endpoint

### PR DESCRIPTION
**Pull request by clintonb**
[ECOM-4087](https://openedx.atlassian.net/browse/ECOM-4087)

### CODE REVIEW

**Comment by clintonb** [comment]
@bderusha @peter-fogg please review

**Comment by bderusha** [comment]
👍

File: course_discovery/apps/api/filters.py [hunk 1]
```diff
@@ -1,9 +1,40 @@
...
-        return get_objects_for_user(request.user, perm)
+        user = request.user
+        username = request.query_params.get('username', None)
+
+        if username:
+            if request.user.is_staff:
+                try:
+                    user = User.objects.get(username=username)
+                except User.DoesNotExist:
+                    raise NotFound('No user with the username [{username}] exists.'.format(username=username))
```
Comment (peter-fogg) [inline course_discovery/apps/api/filters.py]: Should this be translated?

Comment (clintonb) [inline course_discovery/apps/api/filters.py]: Yes. Fixed.

**Comment by peter-fogg** [comment]
:100:

**Pull request merged**